scapy/route6.py

```diff
@@ -200,7 +201,7 @@ def route(self, dst, dev=None):
         # Deal with dev-specific request for cache search
         k = dst
         if dev is not None:
-            k = dst + "%%" + dev
+            k = dst + "%%" + (dev if isinstance(dev, basestring) else dev.pcap_name)
```

Comment:
Oh we don't care: we just need that both read cache and set cache methods use the same thing... I will leave pcap_name